### ISSUE TITLE
h3lis100 Read function issue

### ISSUE THREAD

**Issue by KamranArain** [STMicroelectronics/h3lis100dl#1]
I have a little question that there is 8 bit register at 0x29 for `out x` then why there is need to read reserved address and also it starts from 0x28 instead of 0x29 and there is also right shifting just as 16 bit register. can you please explain it little bit.
``` c
int32_t h3lis100dl_acceleration_raw_get(stmdev_ctx_t *ctx,
                                        int16_t *val)
{
  uint8_t buff[6];
  int32_t ret;

  ret = h3lis100dl_read_reg(ctx, (H3LIS100DL_OUT_X - 1U), buff, 6);
  val[0] = (int16_t)buff[1];
  val[0] = (val[0] * 256) + (int16_t)buff[0];
  val[1] = (int16_t)buff[3];
  val[1] = (val[1] * 256) + (int16_t)buff[2];
  val[2] = (int16_t)buff[5];
  val[2] = (val[2] * 256) + (int16_t)buff[4];

  return ret;
}
```

**Comment by crNewton** [comment]
I have exactly the same question. 

The [datasheet](https://www.st.com/resource/en/datasheet/h3lis100dl.pdf) states that registers 0x28, 0x2A, 0x2C, 2E-2F are reserved. 

_"Registers marked as Reserved must not be changed as they contain the factory calibration
values. Their content is automatically restored when the device is powered up. Writing to
those registers may change calibration data and thus lead to improper functioning of the
device."_

There is no statement about reading those registers, but the datasheet doesn't indicate that those registers should be used to read OUT_ data.

**Comment by avisconti** [comment]
> I have exactly the same question.
> 
> The [datasheet](https://www.st.com/resource/en/datasheet/h3lis100dl.pdf) states that registers 0x28, 0x2A, 0x2C, 2E-2F are reserved.
> 
> _"Registers marked as Reserved must not be changed as they contain the factory calibration values. Their content is automatically restored when the device is powered up. Writing to those registers may change calibration data and thus lead to improper functioning of the device."_
> 
> There is no statement about reading those registers, but the datasheet doesn't indicate that those registers should be used to read OUT_ data.

Yes, it is a typo. Thanks to both of you for noticing it. I'll provide a fix to it. 
Thanks again

**Comment by avisconti** [comment]
Fixed via cab6e09
